src/components/URLImportDialog/index.js

```diff
@@ -0,0 +1,255 @@
+        },
+    },
+    closeDialog: {
+        position: "absolute",
+        right: theme.spacing(1),
+        top: theme.spacing(1),
+        color: theme.palette.blueGrey,
+    },
+    importDefault: {
+        background: theme.palette.primary.main,
+        marginBottom: 25,
+
+        [theme.breakpoints.down("sm")]: {
+            width: "100%", // Easier to hit on mobile.
+            marginTop: 10,
+            marginBottom: 0,
+        },
+    },
+    importError: {
+        background: theme.palette.error.main,
+        marginBottom: 25,
+
+        // Hover colors get weird after switching to error
+        // without this.
+        "&:hover": {
+            backgroundColor: theme.palette.error.dark,
+        },
+
+        [theme.breakpoints.down("sm")]: {
+            width: "100%", // Easier to hit on mobile.
+            marginTop: 10,
+            marginBottom: 0,
+        },
+    },
+}));
+
+const setupEvent = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+};
+
+const validURL = (possibleURL) => {
+    let retval = true;
+
+    try {
+        const u = new URL(possibleURL);
+
+        if (!["http:", "https:", "ftp:"].includes(u.protocol)) {
+            retval = false;
+        }
+    } catch (e) {
+        retval = false;
+    } finally {
+        return retval;
+    }
+};
+
+const URLImportTextField = (props) => {
+    const classes = useStyles();
+    const [uploadURL, setUploadURL] = useState("");
+    const [isValidURL, setIsValidURL] = useState(false);
+    const [hasFirstFocused, setHasFirstFocused] = useState(false);
+
+    const { addURLFn, intl } = props;
+
+    const errored = !isValidURL && hasFirstFocused;
+
+    const clickHandler = (event) => {
+        setupEvent(event);
+        addURLFn(event, uploadURL);
+        setUploadURL("");
+        setIsValidURL(false);
+    };
+
+    return (
+        <div className={classes.container}>
+            <TextField
+                fullWidth
+                id={buildID(ids.BASE_ID, ids.TEXTFIELD)}
+                placeholder={fmt(intl, "placeholder")}
+                helperText={
+                    isValidURL || (!isValidURL && !hasFirstFocused)
+                        ? fmt(intl, "helperText")
+                        : fmt(intl, "helperTextError")
+                }
+                InputLabelProps={{
+                    shrink: true,
+                }}
+                aria-label={fmt(intl, "textFieldAriaLabel")}
+                className={classes.urlField}
+                error={errored}
+                value={uploadURL}
+                onFocus={(e) => {
+                    setHasFirstFocused(true);
+                }}
+                onBlur={(e) => {
+                    // If it's empty, we don't care if it's already been focused on.
+                    if (uploadURL === "") {
+                        setHasFirstFocused(false);
+                    }
+                }}
+                onChange={(e) => {
```

Comment:
We have been using [Formik](https://jaredpalmer.com/formik/) in the current UI to handle user-input forms. It will setup handlers like this for you.

See https://github.com/cyverse-de/ui/blob/master/react-components/src/apps/quickLaunch/CreateQuickLaunchDialog.js for a basic example of a simple 1-field form (well, 1 text field, though it has a checkbox too).

This form is simple enough that this isn't a blocker, and we can merge this PR as-is; but I'd recommend converting this to Formik at least in a future PR.